index.js

```diff
@@ -28,12 +28,27 @@ Toolkit.run(async tools => {
 
   const majorWords = process.env['INPUT_MAJOR-WORDING'].split(',')
   const minorWords = process.env['INPUT_MINOR-WORDING'].split(',')
+  const preReleaseWords = ['pre-alpha', 'pre-beta', 'pre-rc']
+
   let version = 'patch'
+  let foundWord = null;
+  
   if (messages.some(
     message => /^([a-zA-Z]+)(\(.+\))?(\!)\:/.test(message) || majorWords.some(word => message.includes(word)))) {
     version = 'major'
   } else if (messages.some(message => minorWords.some(word => message.includes(word)))) {
     version = 'minor'
+  } else if (messages.some(message => preReleaseWords.some(word => {
+        if (message.includes(word)) {
+          foundWord = word;
+          return true;
+        } else {
+          return false;
+        }
+      }
+    ))) {
+      const preid = foundWord.split("-")[1];
```

Comment:
@gmaggiodev are you missing an `if` here to check if foundWord is defined?
otherwise if it is null, it will fail due to split() on null

Reply:
I skip this of null because if the condition is true (some(... some...)) means that at least one of the preReleaseWords has been found and for all the 3 words there is a "-" in the text. So I'm sure the split will return always 2 items in the "else if" block. Anyway if you prefer we can check it.

Reply:
true! didnt see that :D hard to read in github without checking it out